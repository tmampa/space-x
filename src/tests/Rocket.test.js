import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../src/redux/configureStore'
import RocketItem from '../../src/components/rocketsComponents/RocketItem';

describe('Rocket Element Tests', () => {
  test('Rocket Element renders correctly', () => {
    render(
      <Provider store={store}>
        <RocketItem />
      </Provider>
    );
    expect(screen.queryByText(/Missions/)).toBeNull();
		expect(screen.queryByText(/Description/)).toBeNull();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketItem />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
