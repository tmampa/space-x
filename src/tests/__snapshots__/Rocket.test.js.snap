// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Element Tests renders correctly 1`] = `
<li
  style={
    Object {
      "display": "grid",
      "gap": "2%",
      "gridTemplateColumns": "24% 74%",
      "padding": "1rem 0",
      "position": "relative",
    }
  }
>
  <div>
    <img
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": "1.25rem",
        "gap": "0.75rem",
      }
    }
  >
    <h3 />
    <div
      className="rocket-desc-wrapper"
    />
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "hsl(230deg 90% 60%)",
            "border": "1px solid hsl(230deg 90% 60%)",
            "borderRadius": "0.25rem",
            "color": "hsl(0deg 0% 99%)",
            "cursor": "pointer",
            "fontSize": "1.175rem",
            "padding": "0.35rem",
            "textAlign": "center",
            "whiteSpace": "nowrap",
          }
        }
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</li>
`;
